systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/MultipleListenersST.java

```diff
@@ -0,0 +1,353 @@
+
+                    InternalKafkaClient internalPlainKafkaClient = new InternalKafkaClient.Builder()
+                        .withUsingPodName(kafkaClientsPlainPodName)
+                        .withListenerName(listener.getName())
+                        .withTopicName(topicName)
+                        .withNamespaceName(NAMESPACE)
+                        .withClusterName(CLUSTER_NAME)
+                        .withMessageCount(MESSAGE_COUNT)
+                        .build();
+
+                    LOGGER.info("Checking produced and consumed messages to pod:{}", kafkaClientsPlainPodName);
+
+                    // verify phase
+                    internalPlainKafkaClient.checkProducedAndConsumedMessages(
+                        internalPlainKafkaClient.sendMessagesPlain(),
+                        internalPlainKafkaClient.receiveMessagesPlain()
+                    );
+                }
+            }
+        }
+    }
+
+    private Map<KafkaListenerType, List<GenericKafkaListener>> generateTestCases() {
+
+        LOGGER.info("Starting to generate test cases for multiple listeners");
+
+        int stochasticCount;
+
+        for (KafkaListenerType kafkaListenerType : KafkaListenerType.values()) {
+
+            LOGGER.info("Generating {} listener", kafkaListenerType.name());
+
+            List<GenericKafkaListener> testCaseListeners = new ArrayList<>(5);
+
+            switch (kafkaListenerType) {
+                case NODEPORT:
+                    stochasticCount = ThreadLocalRandom.current().nextInt(2, 5);
+
+                    for (int j = 0; j < stochasticCount; j++) {
+
+                        boolean stochasticCommunication = ThreadLocalRandom.current().nextInt(2) == 0;
+
+                        testCaseListeners.add(new GenericKafkaListenerBuilder()
+                            .withName(generateRandomListenerName())
+                            .withPort(10900 + j)
+                            .withType(KafkaListenerType.NODEPORT)
+                            .withTls(stochasticCommunication)
+                            .build());
+                    }
+                    break;
+                case LOADBALANCER:
+                    stochasticCount = ThreadLocalRandom.current().nextInt(2, 3);
+
+                    for (int j = 0; j < stochasticCount; j++) {
+
+                        boolean stochasticCommunication = ThreadLocalRandom.current().nextInt(2) == 0;
+
+                        testCaseListeners.add(new GenericKafkaListenerBuilder()
+                            .withName(generateRandomListenerName())
+                            .withPort(11900 + j)
+                            .withType(KafkaListenerType.LOADBALANCER)
+                            .withTls(stochasticCommunication)
+                            .build());
+                    }
+                    break;
+                case ROUTE:
+                    testCaseListeners.add(new GenericKafkaListenerBuilder()
```

Comment:
Why not a random number of routes?

Reply:
That's an issue with the multiple routes we spoke with @scholzj (related with DNS) I don't know it is already resolved.

Reply:
It is not related to DNS - this was the bug with unique ports per listener which should be fixed now in Kafka 2.7.0 - you should leave there some comment / TODO to add it later.